gangplank/ocp/cosa-pod.go

```diff
@@ -335,3 +388,159 @@ func streamPodLogs(logging *bool, pod *v1.Pod) error {
 
 	return nil
 }
+
+// encodeToJSON renders the JSON definition of a COSA pod. Podman prefers
+// non-pretty JSON.
+func encodeToJSON(pod *v1.Pod, envVars []v1.EnvVar) (string, error) {
+	scheme := runtime.NewScheme()
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return "", err
+	}
+	s := runtime.NewScheme()
+	e := json.NewSerializerWithOptions(
+		json.DefaultMetaFactory, s, s,
+		json.SerializerOptions{Pretty: false},
+		// json.SerializerOptions{Yaml: true, Pretty: false},
+	)
+
+	var d bytes.Buffer
+	dW := io.Writer(&d)
+
+	if err := e.Encode(pod, dW); err != nil {
+		return "", err
+	}
+
+	return d.String(), nil
+}
+
+// PodmanRunner runs the work in a Podman container using workDir as `/srv`
+// The host must have oci-kvm-hook installed, which ensures that /dev/kvm
```

Comment:
Hmm that's deprecated.  Doesn't it work to request it via the API equivalent of `--device /dev/kvm`?